frontend/packages/dev-console/src/utils/kebab-actions.ts

```diff
@@ -9,20 +9,30 @@ import {
 } from '@console/internal/models';
 import { ModifyApplication, EditApplication } from '../actions/modify-application';
 
-const modifyApplicationRefs = [
+const modifyWebConsoleApplicationRefs = [
   referenceForModel(DeploymentConfigModel),
   referenceForModel(DeploymentModel),
   referenceForModel(DaemonSetModel),
   referenceForModel(StatefulSetModel),
 ];
 
+const modifyApplicationRefs = [
+  referenceForModel(DaemonSetModel),
+  referenceForModel(StatefulSetModel),
+];
+
 export const getKebabActionsForKind = (resourceKind: K8sKind): KebabAction[] => {
   if (!resourceKind) {
     // no common actions
     return [];
   }
 
-  return _.includes(modifyApplicationRefs, referenceForModel(resourceKind))
-    ? [ModifyApplication, EditApplication]
+  return _.includes(modifyWebConsoleApplicationRefs, referenceForModel(resourceKind))
+    ? [
+        ModifyApplication,
+        ...(!_.includes(modifyApplicationRefs, referenceForModel(resourceKind))
```

Comment:
Why would you invert this? You created an array for things that don't support edit instead of an array for things that do.
Please check for kinds D and DC.

Reply:
Added the check for kinds D and DC.